File: backend-laravel/app/Models/Division.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

/**
 * División (tabla `divisiones`, ej. "1a", "2a"). Junto con `Nivel` y
 * `CicloLectivo` arma un `Curso` (nivel + división + ciclo = curso
 * principal, ej. "3° 1a" del ciclo 2026).
 */
class Division extends Model
{
    use SoftDeletes;

    protected $table = 'divisiones';

    protected $primaryKey = 'id_division';

    protected $fillable = [
        'nombre',
    ];

    protected function casts(): array
    {
        return [
            'deleted_at' => 'datetime',
        ];
    }

    public function cursos(): HasMany
    {
        return $this->hasMany(Curso::class, 'division_id', 'id_division');
    }
}

File: app_flutter/lib/models/division.dart
/// Espejo de lo que devuelve `DivisionesController` (backend-laravel) —
/// una división (ej. "1a", "2a"). Junto con `Nivel` y el ciclo lectivo
/// arma un curso (ver `App\Models\Division` en el backend).
class Division {
  const Division({required this.id, required this.nombre});

  factory Division.fromJson(Map<String, dynamic> json) {
    return Division(
      id: json['id_division'] as int,
      nombre: json['nombre'] as String,
    );
  }

  final int id;
  final String nombre;
}
